src/hooks/useRefreshEffect.ts

```diff
@@ -2,18 +2,20 @@ import { FAST_INTERVAL, SLOW_INTERVAL } from 'config/constants'
 import { DependencyList, EffectCallback, useEffect, useMemo } from 'react'
 import useSWR from 'swr'
 
-export function useFastRefreshEffect(effect: EffectCallback, deps?: DependencyList) {
+type BlockEffectCallback = (blockNumber: number) => ReturnType<EffectCallback>
+
+export function useFastRefreshEffect(effect: BlockEffectCallback, deps?: DependencyList) {
   const { data = 0 } = useSWR([FAST_INTERVAL, 'blockNumber'])
   const depsMemo = useMemo(() => [data, ...(deps || [])], [data, deps])
 
   // eslint-disable-next-line react-hooks/exhaustive-deps
-  useEffect(effect, depsMemo)
+  useEffect(effect.bind(null, data), depsMemo)
```

Comment:
Hmm. This current memo isn't right. For `useFastRefreshEffect`, `effect` should only be called when `blockNumber` is changed. However, with the current logic, `effect` would be called if `deps` are changed even though `data` might not be changed.

Reply:
I think it should be 
```
const callback = useCallback((data) => effect(data), [deps])

useEffect(() => callback(data), [data])
```

Reply:
If `deps` changed, the effect should be called again, some effect depends on `account`

Reply:
I think it should be 
```
const callback = useCallback((data) => effect(data), [deps])

useEffect(() => callback(data), [data])
```

Reply:
It wont't run effect when `callback` changes, no?
Maybe a good chance to introduce some react hook testing there 😅

Reply:
It won't because deps of `useEffect` don't have `callback`. The example isn't good as well because it omits the `callback` in dep while using it. But the logic should be similar to this. 

Totally agreed. we should have Unit Test on hooks

Reply:
I think we should stringify the dependencies before use it in memo, otherwise it will create a new array on each render which cause refresheffects to be triggered. I did the fix seems working, can you review @chef-jojo @chefilip ?

Reply:
IMO, I don't prefer stringify especially in this hook, we don't know what kind of `deps` pass into. It won't work for function and circular ref

Reply:
True

Reply:
So we want effect to be called when one of its dependencies changes right?